configure.ac

```diff
@@ -11,7 +11,9 @@ AM_SILENT_RULES([yes])
 
 AC_CONFIG_MACRO_DIR([builds/autoconf/m4])
 AC_CONFIG_SRCDIR([src/data.cpp])
-AC_CONFIG_HEADERS([config.h])
+# We want our public lcf/config.h to look like the cmake one. This tricks autoheader into generating
+# /config.h (which we don't use) and then configure uses our template for lcf/config.h
+AC_CONFIG_HEADERS([autogen/config.h:config.h.in autogen/lcf/config.h:builds/autoconf.config.h.in])
```

Comment:
This should not really be needed, if we `AC_SUBST` the required variables and add the config file to `AC_CONFIG_FILES`
Then this line can be removed to not use config.h at all.

Reply:
Thank you! This was ugly and I couldn't figure out the best way to handle it.

I will also change it to always define the macros to 0 or 1. The old style `#ifdef` way of detecting libraries is broken because you can forget to include the header.